src/main/java/com/checkmarx/jenkins/tools/CheckmarxInstaller.java

```diff
@@ -195,6 +237,44 @@ public Void call() throws IOException {
             return null;
         }
 
+        public static void copyURLToFile(URL source, String proxyStr, String proxyType, File destination, int connectionTimeoutMillis, int readTimeoutMillis) throws IOException {
+            URLConnection connection;
+            if (proxyStr == null) {
+                connection = source.openConnection();
```

Comment:
should we close this connection in finally too?

Reply:
and add it to the try catch / make sure you close the connection

Reply:
![image](https://github.com/jenkinsci/checkmarx-ast-scanner-plugin/assets/110327792/7edca5ef-0180-46b4-96e5-ee1d13ac4c2d)